### PR TITLE
build(jest): upgrade to v26

### PR DESCRIPTION
**Pull request by ianforster-sage**
### Proposed behaviour
This change upgrades some testing library dependencies:
```
@types/jest:
 babel-jest
 jest
```
And repairs some tests that the upgrade caused to fail. 
No codesandbox or screenshots are provided here as this is not a visual change.

### Current behaviour
Currently Carbon has a Jest dependency of v24.9.0.

### Checklist

<del>- [ ] Screenshots are included in the PR</del>
<del>- [ ] Carbon implementation and Design System documentation are congruent</del>
<del>- [ ] All themes are supported</del>
- [x] Commits follow our style guide
- [x] Unit tests added or updated
<del>- [ ] Cypress automation tests added or updated</del>
<del>- [ ] Storybook added or updated</del>
<del>- [ ] Typescript `d.ts` file added or updated</del>

### Additional context
After upgrading `element.focus()` is not setting `document.activeElement`. The fix is to attach a container div to the body for enzyme to render the component in.

### Testing instructions
```
git pull origin FE-2761-jest-upgade
rm -rf node_modules/
npm install
npm test
```

### CODE REVIEW

File: src/__experimental__/components/simple-color-picker/simple-color-picker.spec.js [hunk 1]
```diff
@@ -16,6 +16,32 @@ const colorValues = [
 ];
 const name = 'test-group';
 
+function renderAttached(renderer = mount, props, childProps) {
```
Comment (edleeks87) [inline src/__experimental__/components/simple-color-picker/simple-color-picker.spec.js:19]: Again I think you could get away with just adding `{ attachTo: document.getElementById('enzymeContainer') }` to the `render` function

File: src/__experimental__/components/input/input.spec.js [hunk 2]
```diff
@@ -21,6 +36,20 @@ describe('Input', () => {
     return mount(component);
   };
 
+  const renderAttached = (props, context) => {
```
[the remark this answers is not in the document]
Comment (ianforster-sage) [reply src/__experimental__/components/input/input.spec.js]: Thanks Ed, I should've cleaned this up. I've removed the duplication here, in the `simple-color-picker`, and elsewhere where it was possible to do so.

File: src/components/accordion/accordion.spec.js [hunk 3]
```diff
@@ -238,9 +238,9 @@ describe('Accordion', () => {
 });
 
 describe('AccordionGroup', () => {
-  let wrapper;
+  let wrapper, container;
```
Comment (mkrds) [inline src/components/accordion/accordion.spec.js:241]: One declaration per line

File: src/components/show-edit-pod/__spec__.js [hunk 4]
```diff
@@ -12,10 +12,13 @@ import { StyledLink } from './show-edit-pod.style';
 
 describe('ShowEditPod', () => {
   describe('when the "editing" prop is set on mount', () => {
-    let wrapper;
+    let wrapper, container;
```
Comment (mkrds) [inline src/components/show-edit-pod/__spec__.js:15]: One declaration per line

File: src/components/tabs/tabs.spec.js [hunk 5]
```diff
@@ -524,11 +524,28 @@ describe('Tabs', () => {
   });
 
   describe('Keyboard navigation', () => {
-    let wrapper, replaceSpy;
+    let wrapper, replaceSpy, container;
```
Comment (mkrds) [inline src/components/tabs/tabs.spec.js:527]: One declaration per line

File: package.json [hunk 6]
```diff
@@ -107,7 +107,7 @@
     "husky": "^3.1.0",
     "i18n-js": "http://github.com/fnando/i18n-js/archive/v3.0.0.rc12.tar.gz",
     "identity-obj-proxy": "3.0.0",
-    "jest": "24.9.0",
+    "jest": "26.0.1",
```
Comment (jamime) [inline package.json:110]: IMO this should be `^26.1.0`, we've had issue in the past when running with multiple versions of jest. `jest@26.0.1`actually includes "@jest/core@26.1.0". This happens when we install an outdated version.

**Comment by jamime** [comment]
### QA Notes

We changed the `events.js` file. This is only used by action popover. I've did a quick regression test on `Test/Action Popover` story in

- Chrome
- Safari
- FireFox

This utility was added to support IE11, as we no longer support IE, all of our supported browsers will be using the native `e.path || e.composedPath()`. The re-ordering of the conditions is purely so we can test the click behaviour in enzyme.

**Pull request merged**